tests/test_views.py

```diff
@@ -158,44 +160,145 @@ def test_view_poll_plugin_attached_to_page_should_return_related_page(self):
-        version2.publish(archived.created_by)
-        version3 = version2.copy(version2.created_by)
+        poll1 = PollFactory()
+        # add poll plugin to page
+        add_plugin(placeholder1, "PollPlugin", "en", poll=poll1, template=0)
 
         with self.login_user_context(self.superuser):
-            response = self.client.get(
-                self.get_view_url(
-                    content_type_id=ContentType.objects.get_for_model(poll).pk,
-                    object_id=poll.id,
-                )
-                + "?state=draft_and_published"
-            )
+            url = self.get_view_url(
+                    content_type_id=ContentType.objects.get_for_model(poll1).pk,
+                    object_id=poll1.id,
+                ) + "?state=draft"
+
+            response = self.client.get(url)
 
         self.assertEqual(response.status_code, 200)
-        self.assertIn("querysets", response.context)
+        self.assertEqual(response.context["querysets"][0].count(), 1)
+        self.assertIn(page_content1, response.context["querysets"][0])
+        self.assertNotIn(page_content2, response.context["querysets"][0])
 
-        # queryset should contain page
-        self.assertEqual(response.context["querysets"][0].count(), 2)
-        self.assertNotIn(archived.content, response.context["querysets"][0])
-        self.assertIn(version2.content, response.context["querysets"][0])
-        self.assertIn(version3.content, response.context["querysets"][0])
+    def test_view_published_filter_applied(self):
+        version1 = PageVersionFactory(
+            content__title="test1", content__language=self.language,
+        )
+        version2 = PageVersionFactory(
+            content__title="test2", content__language=self.language, state=PUBLISHED
+        )
 
-    def test_extra_columns(self):
-        extra_column = ExtraColumn(lambda o: "{} test".format(o), "Test column")
+        page_content1 = version1.content
+        page_content2 = version2.content
 
-        with self.login_user_context(self.superuser), patch(
-            "djangocms_references.views.get_extra_columns", return_value=[extra_column]
-        ):
-            response = self.client.get(self.view_url)
+        placeholder1 = PlaceholderFactory(
+            content_type=ContentType.objects.get_for_model(page_content2),
+            object_id=page_content2.id,
+        )
 
-        self.assertEqual(response.context["extra_columns"], [extra_column])
+        poll1 = PollFactory()
+        # add poll plugin to page
+        add_plugin(placeholder1, "PollPlugin", "en", poll=poll1, template=0)
+        add_plugin(placeholder1, "PollPlugin", "en", poll=poll1, template=0)
+
+        with self.login_user_context(self.superuser):
+            url = self.get_view_url(
+                    content_type_id=ContentType.objects.get_for_model(poll1).pk,
+                    object_id=poll1.id,
+                ) + "?state=published"
+
```

Comment:
This doesn't look like it follows AAA, could you remove the space here?

Reply:
@adam-murray  Thanks, updated the test cases to follow AAA